kaitai_utils.py

```diff
@@ -37,16 +39,66 @@ def get_plugin_title_from_metadata():
         return _to_camel_case(metadata["id"])
 
 
-def write_to_json(data_binary: BinaryIO, writer: BufferedWriter, class_type: Type[KaitaiStruct]):
-    """
-    Writes a binary form of JSON string into a BufferedWriter
-
-    @param data_binary: binary data containing the file content
-    @param writer: bufferedWriter to write the binary form of the JSON string to
-    @param class_type: class that contains the parsing to a KaiTai struct
-    @return: JSON string representing contents of data object
-    """
-    writer.write(bytes(to_json_string(data_binary, class_type), "utf-8"))
+class JsonWriter:
+    def __init__(self, writer: BufferedWriter, trace: ExtractionTrace):
+        self.writer = writer
+        self.trace = trace
+
+    @streamable_list
+    def _list_to_dict(self, object_list: List[Any], path: str) -> Generator[
+        tuple[str, Any], None, None]:
+        for value_index, value in enumerate(object_list):
+            yield self._object_to_dict(value, path + f'.[{value_index}]')
+
+    @streamable_dict
+    def _object_to_dict(self, instance: Any, path: str) -> Generator[Dict[str, Any], None, None]:
+        """
+        Recursive helper method that parses an object to a dictionary.
```

Comment:
Other than the word `Recursive`, this pydoc is exactly the same as the pydoc for `_parameters_dict`, but a lot more seems to be happening here. Would it be possible to make this a bit more specific?

Reply:
good point. I fixed it in HANSKEN-21047 since that's the branch we'll be moving forward with. Is that okay?